src/main/java/de/tum/in/www1/artemis/web/rest/metis/ReactionResource.java

```diff
@@ -0,0 +1,144 @@
+        User user = this.userRepository.getUserWithGroupsAndAuthorities();
+        if (reaction.getId() != null) {
+            throw new BadRequestAlertException("A new reaction cannot already have an ID", ENTITY_NAME, "idexists");
+        }
+        final Course course = courseRepository.findByIdElseThrow(courseId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);
+
+        // set user to current user
+        reaction.setUser(user);
+
+        // we query the repository dependent on the type of posting
+        Reaction savedReaction;
+        if (posting instanceof Post) {
+            Post post = postRepository.findByIdElseThrow(posting.getId());
+            reaction.setPost(post);
+            savedReaction = reactionRepository.save(reaction);
+            post.addReaction(reaction);
+            postRepository.save(post);
+        }
+        else {
+            AnswerPost answerPost = answerPostRepository.findByIdElseThrow(posting.getId());
+            reaction.setAnswerPost(answerPost);
+            savedReaction = reactionRepository.save(reaction);
+            answerPost.addReaction(reaction);
+            answerPostRepository.save(answerPost);
+        }
+
+        return ResponseEntity.created(new URI("/api/courses/" + courseId + "/postings/reactions/" + savedReaction.getId())).body(savedReaction);
+    }
+
+    /**
+     * DELETE /courses/{courseId}/posts/{id}/reactions/{reactionId} : delete the reaction with {id}.
+     *
+     * @param courseId   course the posting that is reacted on belongs to
+     * @param reactionId the id of the reaction to delete
+     * @return the ResponseEntity with status 200 (OK) or 400 (Bad Request) if the data is inconsistent
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @DeleteMapping("courses/{courseId}/postings/reactions/{reactionId}")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Void> deleteReaction(@PathVariable Long courseId, @PathVariable Long reactionId) throws URISyntaxException {
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        courseRepository.findByIdElseThrow(courseId);
+        Reaction reaction = reactionRepository.findByIdElseThrow(reactionId);
+
+        // check if user that wants to delete reaction is user that created the reaction
+        if (!user.equals(reaction.getUser())) {
+            throw new AccessForbiddenException("Reaction", reaction.getId());
+        }
+
+        String entity = "";
+        if (reaction.getPost() != null) {
+            entity = "post with id: " + reaction.getPost().getId();
+        }
+        else if (reaction.getAnswerPost() != null) {
+            entity = "answer post with id: " + reaction.getAnswerPost().getId();
+        }
+        log.info("Reaction deleted by " + user.getLogin() + ". Reaction: " + reaction.getEmojiId() + " for " + entity);
```

Comment:
Maybe you could use log.debug instead of log.info here? I don't know how important it would be to see this log on production.

Reply:
ah good catch! totally agree... I change this